### ISSUE TITLE
IPv4 options

### ISSUE THREAD

**Issue by jhsmt** [sonic-net/DASH#468]
This is more a clarification question:
The IPv4 headers (and parser) include IPv4 options. Is the motivation to look at IP options driven by desire for correctness? If yes, what about TCP options (which seem to be missing)? It is a little unusual to have IP options in data centre traffic but otoh very common to have TCP options.

**Comment by KrisNey-MSFT** [comment]
Will take a look and get back re: this.  @sharmasushant 
Do we want a DASH device to be able to look past IPv4 options (to ask another way).

**Comment by jafingerhut** [comment]
For example, even if a DASH device does not need to pay careful attention to the _contents_ of IPv4 options, if present, is it important that a DASH device be able to _skip over_ IPv4 options in order to find the L4 header, and if it is TCP or UDP, extract the source & destination port from the packet in order to perform DASH ACL classification, and/or matching the packet against a cached flow record in the fast path.

**Comment by jhsmt** [comment]
On Wed, Dec 6, 2023 at 12:07 PM Andy Fingerhut ***@***.***>
wrote:

> For example, even if a DASH device does not need to pay careful attention
> to the *contents* of IPv4 options, if present, is it important that a
> DASH device be able to *skip over* IPv4 options in order to find the L4
> header, and if it is TCP or UDP, extract the source & destination port from
> the packet in order to perform DASH ACL classification, and/or matching the
> packet against a cached flow record in the fast path.
>
>
Assuming that DASH does not care about anything beyond the transport header
or any transport header options then this is a reasonable explanation.

**Comment by r12f** [comment]
@jhsmt , yes. in the current supported scenarios, such as VNET, PrivateLink and etc, the TCP options are not used, hence not parsed.

**Comment by jhsmt** [comment]
On Thu, Dec 7, 2023 at 6:10 PM Riff ***@***.***> wrote:

> @jhsmt <https://github.com/jhsmt> , yes. in the current supported
> scenarios, such as VNET, PrivateLink and etc, the TCP options are not used,
> hence not parsed.
>

Ok, I think we can close this issue then.

**Comment by chrispsommers** [comment]
Closed per author's advice.
